src/core/net/ip6.cpp

```diff
@@ -627,6 +629,23 @@ ThreadError Ip6::ProcessReceiveCallback(const Message &aMessage, const MessageIn
     mReceiveIp6DatagramCallback(messageCopy, mReceiveIp6DatagramCallbackContext);
 
 exit:
+
+    switch (error)
+    {
+    case kThreadError_NoBufs:
```

Comment:
`kThreadError_NoBufs` is the error when cloning the message fails from line 625.. (few lines above this)...
```
VerifyOrExit((messageCopy = aMessage.Clone()) != NULL, error = kThreadError_NoBufs);
```